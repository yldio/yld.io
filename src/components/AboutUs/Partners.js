import React from 'react'
import { Padding } from 'styled-components-spacing'

import { Grid } from '../grid'
import { SectionTitle } from '../Typography'
import LogoGrid from '../Common/LogoGrid'

const Partners = ({ title, partners }) => (
  <Grid>
    <Padding
      top={{ smallPhone: 3, tablet: 4 }}
      bottom={{ smallPhone: 3.5, tablet: 5 }}
    >
      <Padding bottom={{ smallPhone: 3, tablet: 4 }}>
        <SectionTitle>{title}</SectionTitle>
      </Padding>
      <LogoGrid companies={partners} />
    </Padding>
  </Grid>
)

export default Partners
