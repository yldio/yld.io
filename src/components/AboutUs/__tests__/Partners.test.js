import React from 'react'
import renderComponent from '../../../utils/tests/renderComponent'
import Partners from '../Partners'

describe('ClientTestimonial', () => {
  it('renders correctly', () => {
    const title = 'Technology partnerships'
    const partners = [
      {
        id: 0,
        title: 'nodeJS',
        url: 'https://foundation.nodejs.org/',
        image: {
          file: {
            url: 'img1.jpg'
          }
        }
      },
      {
        id: 1,
        title: 'aws',
        url: 'https://aws.amazon.com/pt/',
        image: {
          file: {
            url: 'img2.jpg'
          }
        }
      },
      {
        id: 2,
        title: 'another partner',
        image: {
          file: {
            url: 'img3.jpg'
          }
        }
      }
    ]
    const tree = renderComponent(<Partners title={title} partners={partners} />)

    expect(tree).toMatchSnapshot()
  })
})
