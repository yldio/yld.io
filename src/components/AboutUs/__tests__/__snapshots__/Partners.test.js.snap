// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientTestimonial renders correctly 1`] = `
<div
  className="sc-htpNat toUCL"
>
  <div
    className="sc-bwzfXH egEeOi"
  >
    <div
      className="sc-bwzfXH bXntSd"
    >
      <h2
        className="sc-EHOje hObgUw"
      >
        Technology partnerships
      </h2>
    </div>
    <div
      className="sc-ifAKCX dJZmHe"
    >
      <div
        className="sc-gqjmRU jgkcLp sc-bxivhb bvhjTU"
        width={
          Array [
            0.5,
            0.5,
            0.5,
            0.5,
            0.3333333333333333,
            0.25,
          ]
        }
      >
        <a
          href="https://foundation.nodejs.org/"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
          title="nodeJS"
        >
          <img
            className="sc-gZMcBi eFAVvK"
            src="img1.jpg"
            style={
              Object {
                "filter": "grayscale(1)",
                "saturate": "0",
              }
            }
            width="250px"
          />
        </a>
      </div>
      <div
        className="sc-gqjmRU jgkcLp sc-bxivhb bvhjTU"
        width={
          Array [
            0.5,
            0.5,
            0.5,
            0.5,
            0.3333333333333333,
            0.25,
          ]
        }
      >
        <a
          href="https://aws.amazon.com/pt/"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
          title="aws"
        >
          <img
            className="sc-gZMcBi eFAVvK"
            src="img2.jpg"
            style={
              Object {
                "filter": "grayscale(1)",
                "saturate": "0",
              }
            }
            width="250px"
          />
        </a>
      </div>
      <div
        className="sc-gqjmRU jgkcLp sc-bxivhb bvhjTU"
        width={
          Array [
            0.5,
            0.5,
            0.5,
            0.5,
            0.3333333333333333,
            0.25,
          ]
        }
      >
        <img
          className="sc-gZMcBi dhwgne"
          src="img3.jpg"
          style={
            Object {
              "filter": "grayscale(1)",
              "saturate": "0",
            }
          }
          title="another partner"
        />
      </div>
    </div>
  </div>
</div>
`;
